### PR TITLE
Be/feature/authentication

### PR DESCRIPTION
**Pull request by bgmrsln**
Added authentication related endpoints

### CODE REVIEW

**Comment by rburaksaritas** [comment]
Is it possible for you to rename Services/authentication.py to Services/authentication_service.py for the sake of file naming consistency?

**Comment by mehmetkuzu** [comment]
Reviewing now

**Comment by bgmrsln** [comment]
> Is it possible for you to rename Services/authentication.py to Services/authentication_service.py for the sake of file naming consistency?

Yes, you are right

File: Disaster-Response-Platform/backend/Controllers/user_controller.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+       #todo what should itreturn, my brain does not process anymore
+        return {"status": "OK"}
+  
+    except Exception as e:
+        # Handle any other unexpected exceptions here
+        raise HTTPException(status_code=status.HTTP_500_INTERNAL_SERVER_ERROR, detail="An unexpected error occurred")
+
+# Login route
+@router.post("/token", response_model=user_model.Token)
```
Comment (mervegrbz) [inline Disaster-Response-Platform/backend/Controllers/user_controller.py:55]: can you change the name as sign-in :)

File: Disaster-Response-Platform/backend/Services/authentication_service.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+from jose import JWTError, jwt
+from passlib.context import CryptContext
+from Models import user_model
+from typing import Annotated
+
+db = MongoDB.getInstance()
+pwd_context = CryptContext(schemes=["bcrypt"], deprecated="auto")
+
+
+SECRET_KEY = "your-secret-key" #todo: put it to config
```
Comment (mervegrbz) [inline Disaster-Response-Platform/backend/Services/authentication_service.py:16]: Will you do this on config?

**Review by mervegrbz — CHANGES_REQUESTED** [review]

File: Disaster-Response-Platform/backend/Controllers/user_controller.py (hunk 1 above)
Comment (bgmrsln) [reply Disaster-Response-Platform/backend/Controllers/user_controller.py:55]: Makes the most sense

**Review by mehmetkuzu — APPROVED** [review]
New authentication

File: Disaster-Response-Platform/backend/Controllers/user_controller.py (hunk 1 above)
Comment (mehmetkuzu) [reply Disaster-Response-Platform/backend/Controllers/user_controller.py]: Now it's ok. It seems these changes are done.

**Review by mervegrbz — APPROVED** [review]

**Pull request merged**